test/e2e/e2e_test.go

```diff
@@ -1186,18 +1170,6 @@ var _ = Describe("e2e non-vxlan external gateway and update validation", func()
 		if err != nil {
 			framework.Failf("failed to add the pod route on the test container: %v", err)
 		}
-		// Wait for the exGW pod networking to be almost, updated
-		wait.PollImmediate(1*time.Second, 30*time.Second, func() (bool, error) {
-			jsonFlag := "jsonpath='{.metadata.annotations.k8s\\.ovn\\.org/hybrid-overlay-external-gw}'"
```

Comment:
how did this ever pass when we dont use hybrid overlay here?

Reply:
because if you don't check for the error of ` wait.PollImmediate`  this  never fails, is just a loop with retries

see comment https://github.com/ovn-org/ovn-kubernetes/pull/1725/files#r492389439